draft-yasskin-http-origin-signed-responses.md

```diff
@@ -573,6 +573,110 @@ Signature: sig3;
 https://example.com/resource.validity could also expand the set of signatures if
 its `signatures` array contained more than 2 elements.
 
+## The Accept-Signature header ## {#accept-signature}
+
+`Signature` header fields cost on the order of 300 bytes for ECDSA signatures,
+so servers might prefer to avoid sending them to clients that don't intend to
+use them. A client can send the `Accept-Signature` header field to indicate that
```

Comment:
When would a client send this? Would you expect a UA to always send this header?

Reply:
I would expect the AMP crawler to always send it. A browser might omit it on normal requests, but send it when there's an [`integrity="ed25519-xyz"` attribute](https://github.com/mikewest/signature-based-sri#the-proposal), and, when the user says "save page as", it might try re-requesting all the resources with the header to see if it can include proofs that the resources aren't modified. Or if cross-signatures become a thing, the browser might always send a version of the header that asks for the cross-signature it cares about.

Reply:
I'd expect UA implementing pre-fetch mechanisms (incl. Chromium) to always send this header.
LINK rel={prefetch,preload} primitives may prefetch the resources that are later meant to be integrity checked. While re-sending the request on <script integrity> usage is theoretically possible, it would incur additional latency which was the whole point of the prefetch.

Reply:
We wouldn't need to pass the header for [`as`](https://html.spec.whatwg.org/multipage/semantics.html#attr-link-as) types that don't support [`integrity` attributes](https://w3c.github.io/webappsec-subresource-integrity/#interface-extensions), but yeah, that's a good point for preloaded scripts and stylesheets. @otherdaniel, would it make sense to put `integrity` attributes on prefetch links? [`<link>` already supports the attribute](https://html.spec.whatwg.org/multipage/semantics.html#attr-link-integrity) for the benefit of stylesheets.

Servers can also opt to send a Signature unconditionally when they expect it to be needed by their SRI directives, although we probably don't want to risk that performance footgun.

Reply:
Regarding _integrity attributes on prefetch links_: We've been [discussing this](https://bugs.chromium.org/p/chromium/issues/detail?id=677022) in a different context, and the consensus then was that from a web developer's perspective it makes little sense to require integrity attributes on prefetches. The present Chrome/Chromium implementation doesn't deal super well with this, but we thought our implementation choices shouldn't be the deciding factor.

I think one can argue this either way, but my gut feeling is that the prefetch shouldn't be burdened with this.

Reply:
'k, I realized that `ed25519key; *` already matches all keys, so the browser can use that on prefetches if we don't want those to include `integrity` attributes. I've added an example for this. Or servers can send their `Signature` headers unconditionally for those resources.

Reply:
This is a bit of a nitpick, but: As I read draft-ietf-httpbis-header-structure-02, a lone asterisk wouldn't match the grammar given there. It specifies 

>  binary = "*" 1\*21846(base64)

which I read as at least 1 and at most 21846 base64 characters. I guess there should be some way to specifiy an empty binary string, but I can't figure out how.